SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUsagePoller.java

```diff
@@ -126,15 +126,35 @@ public void runActionOnPoll() {
     CompletableFutures.allOf(usageFutures).join();
 
     usageManager.saveClusterUtilization(
-        getClusterUtilization(utilizationPerRequestId, totalMemBytesUsed.get(), totalMemBytesAvailable.get(), totalCpuUsed.get(), totalCpuAvailable.get(), totalDiskBytesUsed.get(), totalDiskBytesAvailable
-            .get(), now));
+        getClusterUtilization(
+            utilizationPerRequestId, totalMemBytesUsed.get(), totalMemBytesAvailable.get(),
+            totalCpuUsed.get(), totalCpuAvailable.get(), totalDiskBytesUsed.get(), totalDiskBytesAvailable.get(), now));
     utilizationPerRequestId.values().forEach(usageManager::saveRequestUtilization);
 
     if (configuration.isShuffleTasksForOverloadedSlaves()) {
       shuffleTasksOnOverloadedHosts(overLoadedHosts);
     }
   }
 
+  public CompletableFuture<SingularitySlaveUsage> getSlaveUsage(SingularitySlave slave) {
```

Comment:
The only thing I find as a code smell here is that now the offer scoring flow will rely on the usage semaphore and executor having enough permits/threads. Since within the offer scoring we are already in a block that is executed async, it may be worth calling `collectSlaveUsage` directly when used from the offer context, to avoid an additional layer of async work